src/FacetedSearchContainer.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import _ from 'lodash'
+
+import FilterSidebar from './FilterSidebar'
+import FilterList from './FilterList'
+
+import {ResultPropTypes} from './ResultPropTypes'
+
+class FacetedSearchContainer extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      facets:
+        _(props.results)
+          .flatMap(`facets`)
+          .compact()   // lodash will emit undefined if `facets` doesn’t exist :/
+          .uniqWith(_.isEqual)
+          .map((facet) => ({...facet, disabled: false}))
+          .value(),
+      selectedFacets: {}  // TODO (?) Build initial state of checked filters from props if wrapped in React Router
+    }
+
+    this._handleChange = this._handleChange.bind(this)
+  }
+
+  _filterResults(facets) {
+    return this.props.results.filter((result) =>
+      // Leave the results that, for every group, have at least one matching property within that group
+      // Note: every returns true for empty arrays
+      Object.entries(facets)
+        .every(
+          ([facetGroup, facetValues]) =>
+            result.facets.some(
+              (resultFacet) =>
+                facetGroup === resultFacet.group && facetValues.map(f => f.value).includes(resultFacet.value))))
+  }
+
+  _hasNoResults(selectedFacets, additionalFacet) {
+    const mergedFacets =
+      _.mergeWith(
+        _.defaultsDeep({}, selectedFacets),
+        {[additionalFacet.group]: additionalFacet},
+        (objValue, srcValue) => _.uniq((objValue || []).concat(srcValue))
```

Comment:
What is an `obj` and what is a `src`? I find this block of code particularly difficult to understand...

Reply:
It’s a generic callback passed to `mergeWith`. I used the same notation that Lodash does. I’ll try to think a bit about using better names before merging.